docs-en/chapter_hashing/hash_collision.md

```diff
@@ -1,45 +1,45 @@
 # Hash Collision
 
-As mentioned in the previous section, **usually the input space of a hash function is much larger than its output space**, making hash collisions theoretically inevitable. For example, if the input space consists of all integers and the output space is the size of the array capacity, multiple integers will inevitably map to the same bucket index.
+The previous section mentioned that, **in most cases, the input space of a hash function is much larger than the output space**, so theoretically, hash collisions are inevitable. For example, if the input space is all integers and the output space is the size of the array capacity, then multiple integers will inevitably be mapped to the same bucket index.
 
-Hash collisions can lead to incorrect query results, severely affecting the usability of hash tables. To solve this problem, we expand the hash table whenever a hash collision occurs, until the collision is resolved. This method is simple and effective but inefficient due to the extensive data transfer and hash value computation involved in resizing the hash table. To improve efficiency, we can adopt the following strategies:
+Hash collisions can lead to incorrect query results, severely impacting the usability of the hash table. To address this issue, whenever a hash collision occurs, we perform hash table resizing until the collision disappears. This method is simple, straightforward, and effective, but its efficiency is too low because hash table resizing requires a large amount of data movement and hash value calculation. To improve efficiency, we can adopt the following strategies:
```

Comment:
Recommend changing to "This approach is pretty simple, straightforward, and working well." I would argue the usage of "effective" here as the following sentence indicates this approach is ineffective. The next sentence can start with "However, "